elastic/kibana
# [Infra UI] Only refresh the metadata section not the whole page when retrying metadata fetch

Closes #161367
## Summary

This PR changes the metadata error callout to re-fetch the metadata instead of reloading the whole page. As the components are now used also in the asset details view refreshing the flyout won't be needed - as also it won't help the user if we fetch other data again when the metadata is missing. The current logic will make the request to the metadata again in case of an error. This way the other content will stay the same as we don't need to reload the whole flyout/tab in case the metadata request fails

## Testing
I couldn't find a good way to 'fake' the error happening only once ( and getting a result after the refetch) I did 2 things trying to simulate the behavior (not ideal but it's fine for checking the cases)

1. Change the [metadata endpoint name](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/components/asset_details/hooks/use_metadata.ts/#L32C8-L32C8) (to metadataaa for example) to cause 404 error (or comment the  parameter to get 500)
 - Check the error callout
 

https://github.com/elastic/kibana/assets/14139027/fbf833c7-2f94-4027-bf58-3dbc8d95a090

2. Change the metadata hook to send an error initially and to get the actual metadata error value when the re-fetch is triggered: 
 
<img width="808" alt="fake_error" src="https://github.com/elastic/kibana/assets/14139027/b81c3c98-d957-4c0b-8f2e-676c26a0e916">


https://github.com/elastic/kibana/assets/14139027/d65f5c3e-9d81-4fec-a0bf-338ef50ca7db
